Add anti-affinity for envoy deployed by provisioner (#6148)
projectcontour/contour: changelogs/unreleased/6148-lubronzhan-minor.md

## Add anti-affinity rule for envoy deployed by provisioner

> Now by default, the envoy deployment created by gateway provistioner will include an default anti-affinity rule. Also the anti-affinity rule in [default envoy deployment manifest](../../examples/deployment/03-envoy-deployment.yaml) is updated to `preferredDuringSchedulingIgnoredDuringExecution` to be consistent with contour deployment.

Reviewer: Little wordsmithing on the CL:

```suggestion
The envoy deployment created by the gateway provisioner now includes a default anti-affinity rule. The anti-affinity rule in the [example envoy deployment manifest](https://github.com/projectcontour/contour/blob/main/examples/deployment/03-envoy-deployment.yaml) is also updated to `preferredDuringSchedulingIgnoredDuringExecution` to be consistent with the contour deployment and the gateway provisioner anti-affinity rule.
```
